frappe/core/doctype/user_permission/user_permission.py

```diff
@@ -9,6 +9,15 @@
 from frappe import _
 
 class UserPermission(Document):
+	def validate(self):
+		exists = frappe.db.get_all(self.doctype, filters={
```

Comment:
You can also use `frappe.db.exists` (it takes filters too!)

If you use get_all, best to set `limit = 1`